DigitalLearningSolutions.Data/Services/RecommendedLearningService.cs

```diff
@@ -67,40 +67,111 @@ int delegateId
+                competencyLearningResources.Where(clr => clr.LearningHubResourceReferenceId == rr.RefId)
+                    .ToList();
 
             var competencyResourceAssessmentQuestionParameters =
                 competencyLearningResourcesDataService
-                    .GetCompetencyResourceAssessmentQuestionParameters(clrsForResource.Select(clr => clr.Id)).ToList();
+                    .GetCompetencyResourceAssessmentQuestionParameters(clrsForResource.Select(clr => clr.Id))
+                    .ToList();
+
+            return AreDelegateAnswersWithinRangeToDisplayResource(
+                clrsForResource,
+                competencyResourceAssessmentQuestionParameters,
+                selfAssessmentId,
+                delegateId
+            )
+                ? new RecommendedResource(
+                    learningHubResourceReferenceId,
+                    rr,
+                    incompleteLearningLogItem,
+                    learningLogItemsForResource.Any(ll => ll.CompletedDate != null),
+                    CalculateRecommendedLearningScore(
+                        rr,
+                        clrsForResource,
+                        competencyResourceAssessmentQuestionParameters,
+                        selfAssessmentId,
+                        delegateId
+                    )
+                )
+                : null;
+        }
+
+        private bool AreDelegateAnswersWithinRangeToDisplayResource(
+            List<CompetencyLearningResource> clrsForResource,
+            List<CompetencyResourceAssessmentQuestionParameter> competencyResourceAssessmentQuestionParameters,
+            int selfAssessmentId,
+            int delegateId
+        )
+        {
+            foreach (var competencyLearningResource in clrsForResource)
+            {
+                var delegateResults = selfAssessmentDataService
+                    .GetSelfAssessmentResultsForDelegateSelfAssessmentCompetency(
+                        delegateId,
+                        selfAssessmentId,
+                        competencyLearningResource.CompetencyId
+                    ).ToList();
+
+                var competencyResourceAssessmentQuestionParameter =
+                    competencyResourceAssessmentQuestionParameters.Single(
+                        qp => qp.CompetencyLearningResourceId == competencyLearningResource.Id
+                    );
 
+                var latestConfidenceResult = delegateResults
+                    .Where(
+                        dr => dr.AssessmentQuestionId ==
+                              competencyResourceAssessmentQuestionParameter.AssessmentQuestionId
+                    )
+                    .OrderByDescending(dr => dr.DateTime).FirstOrDefault();
+
+                if (competencyResourceAssessmentQuestionParameter.MinResultMatch <= latestConfidenceResult?.Result &&
+                    latestConfidenceResult?.Result <= competencyResourceAssessmentQuestionParameter.MaxResultMatch)

```

Comment:
The second null conditional operator `?` on latestConfidenceResult is unnecessary as it should be tackled by the first.